relayer/relays/beacon/header/syncer/scale/beacon_scale.go

```diff
@@ -294,3 +294,20 @@ type BeaconState struct {
 	CompactBeaconState
 	BlockRoot types.H256
 }
+
+type VersionedExecutionPayloadHeader struct {
+	Capella *ExecutionPayloadHeaderCapella
+	Deneb   *ExecutionPayloadHeaderDeneb
+}
+
+func (v VersionedExecutionPayloadHeader) Encode(encoder scale.Encoder) error {
+	var err error
+	if v.Capella != nil {
+		encoder.PushByte(0)
+		err = encoder.Encode(v.Capella)
+	} else if v.Deneb != nil {
+		encoder.PushByte(1)
+		err = encoder.Encode(v.Deneb)
+	}
```

Comment:
We should have an else here that sets `err` if no branch is hit.

Reply:
https://github.com/Snowfork/snowbridge/pull/981/commits/9673b21dfa790eb9a4fc6d3ea2ea50e02e2e9196